bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/FrontEnd.java

```diff
@@ -18,314 +18,59 @@
-                "      const currentTime = new Date().toLocaleString(); \n" +
-                "      const fileName = `Profiler_Result_${currentTime}.html`;" +
-                "      anchor.setAttribute(\"download\", fileName);\n" +
-                "\n" +
-                "      // Set the href attribute to the URL of the Blob object\n" +
-                "      anchor.setAttribute(\"href\", URL.createObjectURL(blob));\n" +
-                "\n" +
-                "      // Click the anchor element to download the file\n" +
-                "      anchor.click();\n" +
-                "    }\n" +
-                "  };\n" +
-                "  xhr.send();\n" +
-                "}\n" +
-                "      // Define a function to reset the zoom on the flame graph\n" +
-                "      function resetZoom() {\n" +
-                "         flameGraph.resetZoom();\n" +
-                "      }\n" +
-                "\n" +
-                "      // Define a function to logs a message to the console\n" +
-                "      function onClick(d) {\n" +
-                "         console.info(\"Clicked on \" + d.data.name);\n" +
-                "      }\n" +
-                "\n" +
-                "   </script>\n" +
+    public String readFileAsString() throws IOException {
+        InputStreamReader inputStreamReader;
+        BufferedReader br = null;
+        StringBuilder sb = new StringBuilder();
 
-                "   <script>\n" +
-                "      const myDiv = document.getElementById('details');\n" +
-                "\n" +
-                "      new MutationObserver(() => {\n" +
-                "         const currentText = myDiv.textContent;\n" +
-                "         if (currentText.includes('samples')) {\n" +
-                "            const newText = currentText.replace('samples', 'ms');\n" +
-                "            myDiv.textContent = newText;\n" +
-                "         }\n" +
-                "      }).observe(myDiv, { childList: true });\n" +
-                "   </script>\n" +
+        try (InputStream inputStream = getClass().getClassLoader().getResourceAsStream(FILE_LOCATION)) {
+            if (inputStream == null) {
+                throw new ProfilerRuntimeException("resource file not found: " + FILE_LOCATION);
+            }
+            inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
+            br = new BufferedReader(inputStreamReader);
+            String content = br.readLine();
+            if (content == null) {
+                return sb.toString();
+            }
+            sb.append(content);
 
-                "</body>\n" +
-                "\n" +
-                "</html>";
+            while ((content = br.readLine()) != null) {
+                sb.append('\n').append(content);
+            }
+        } finally {
+            if (br != null) {
+                br.close();
```

Comment:
we need to close `inputStreamReader` also. It will be easy if we can move this and `br` also to the `try` block